sorting/bubble_sort.c

```diff
@@ -1,55 +1,73 @@
-// sorting of array list using bubble sort
+/**
+ * @file
+ * @brief [Bubble sort algorithm
+ * implements](https://en.wikipedia.org/wiki/Bubble_sort)
+ */
+#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 
-/*Displays the array, passed to this method*/
+/**
+ * Display elements of array
+ * @param arr array to be display
+ * @param n length of array
+ */
 void display(int *arr, int n)
 {
-    int i;
-    for (i = 0; i < n; i++)
+    for (int i = 0; i < n; i++)
     {
         printf("%d ", arr[i]);
     }
-
     printf("\n");
 }
 
-/*Swap function to swap two values*/
+/**
+ * Swap two values by using pointer
+ * @param first first pointer of first number
+ * @param second second pointer of second number
+ */
 void swap(int *first, int *second)
 {
     int temp = *first;
     *first = *second;
     *second = temp;
 }
 
-/*This is where the sorting of the array takes place
- arr[] --- Array to be sorted
- size --- Array Size
+/**
+ * Bubble sort algorithm implements
+ * @param arr array to be sorted
+ * @param size size of array
  */
 void bubbleSort(int *arr, int size)
 {
     for (int i = 0; i < size - 1; i++)
     {
+        bool swapped = false;
         for (int j = 0; j < size - 1 - i; j++)
         {
             if (arr[j] > arr[j + 1])
             {
                 swap(&arr[j], &arr[j + 1]);
+                swapped = true;
             }
         }
+        if (!swapped)
+        {
+            break;
```

Comment:
```suggestion
            /* since no more updates we made, the array is already sorted 
                this is an optimization for early termination */
            break;
```